build.gradle.kts

```diff
@@ -223,10 +227,29 @@ tasks {
 		options.compilerArgs.add("-Xlint:all,-exports")
 	}
 
+	// Prepares test-related JVM args
+	val moduleName = "org.junitpioneer"
+	val targetModule = if (modularBuild.toBoolean()) moduleName else "ALL-UNNAMED"
+	val patchModuleArg = "--patch-module=$moduleName=${compileJava.get().destinationDirectory.asFile.get().path}"
+	val testJvmArgs = listOf(
+			"-XX:+IgnoreUnrecognizedVMOptions",
+			"--add-opens=java.base/java.util=$targetModule",
+			"--add-opens=java.base/java.lang=$targetModule",
+			patchModuleArg
+	)
+
 	compileTestJava {
 		options.encoding = "UTF-8"
 		options.compilerArgs.add("-Werror")
-		options.compilerArgs.add("-Xlint:all")
+		var xlintArg = "-Xlint:all"
+		if (modularBuild.toBoolean()) {
+			options.compilerArgs.add(patchModuleArg)
```

Comment:
I know it's a redundant check, but can you pull this line into a separate `if`, so the linting and patching isn't intermingled?